security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -553,9 +557,18 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
 			if sc.isTokenExpired() {
 				cacheLog.Debugf("%s token expired", logPrefix)
-				sc.callbackWithTimeout(connKey, nil /*nil indicates close the streaming connection to proxy*/)
-
-				return true
+				if fetchToken != nil {
```

Comment:
Is it okay for you to add an test with short lived token, and verify that the token refresh logic works as expected?